File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.cpp
class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int prev = 1;
        
        for (int i = 1; i < arr.size(); i++){
            if(arr[i] >= prev + 1){
                prev++;
            }
        }
        return prev;
    }
};

File: 1846-maximum-element-after-decreasing-and-rearranging/1846-maximum-element-after-decreasing-and-rearranging.java
class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        Arrays.sort(arr);
        
        int prev = 0;
        
        for(int i = 0; i< arr.length; i++){
            prev = Math.min(prev + 1, arr[i]);
            
        }
        
        return prev;
    }
}

